specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2019-05-01/examples/NasCluster/Get.json

```diff
@@ -0,0 +1,27 @@
+{
+  "parameters": {
+    "api-version": "2019-05-01",
+    "subscriptionId": "72526b2b-4b61-4700-8f0c-7bb7ae9b6a2d",
+    "resourceGroupName": "System.redmond",
+    "location": "redmond",
+    "nasCluster": "bc489b66-faa7-41f0-bee7-bcba258f10fb"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/72526b2b-4b61-4700-8f0c-7bb7ae9b6a2d/resourceGroups/System.redmond/providers/Microsoft.Fabric.Admin/fabricLocations/redmond/nasClusters/bc489b66-faa7-41f0-bee7-bcba258f10fb",
+        "name": "redmond/bc489b66-faa7-41f0-bee7-bcba258f10fb",
```

Comment:
>bc489b66-faa7-41f0-bee7-bcba258f10fb [](start = 25, length = 36)

Are you going to generate random resource names?

Reply:
Yes. It is a GUID stored in our DB

Reply:
Can we not make it a GUID? GUIDs in resource names are generally a source of customer dissatisfaction.

Reply:
In our internal discussion, we are worry about the resource name is not unique so create such guid to avoid that and prevent user input in AddCluster is not valid for url

Reply:
You can have resources that have names that are unique globally (within a cloud) or within a region. Storage account is one such example that enforces uniqueness. Resources of this type typically come with a checkNameAvailability API to make it easier to assign a name to them. (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#check-name-availability-requests)

Per the RPC (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md), resource names "cannot include: '<', '>', '%', '&', ':', '\', '?', '/' OR any control characters. The max length is 260 characters. All other characters are allowed. The RP is expected to be more restrictive and have its own validation."

Reply:
The problem is customer is not create Resource from ARM or AzureRM. It is by another command in Azure Stack Hub PEP. So the check can't be always the same as ARM. And we are not worry about the URL has guid in this case as customer would not use AzureRM to create/get the resource